articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md

```diff
@@ -0,0 +1,178 @@
+    redirectUri: location.origin
+};
+```
+
+## More Information
+
+### *msal.js*
+
+MSAL is the library used to sign in users and request tokens used to access an API protected by Microsoft Azure Active Directory (Azure AD). The quickstart's *index.html* contains a reference to the library:
+
+```html
+<script src="https://secure.aadcdn.microsoftonline-p.com/lib/0.1.1/js/msal.min.js"></script>
+````
+
+Alternatively, you can download it through npm:
+
+```batch
+npm install msal
+```
+
+### MSAL initialization
+
+The quickstart code also shows how you how you initialize the library:
+
+```javascript
+var userAgentApplication = new Msal.UserAgentApplication(msalconfig.clientID, null, loginCallback, {
+    redirectUri: msalconfig.redirectUri
+});
+```
+
+> |Where  |  |
+> |---------|---------|
+> |`ClientId`     |Application Id from the application registered in the Azure Portal|
+> |`authority`    |It is the authority URL. *null* sets the default authority to <i>https:<span/>//login.microsoftonline.com/common</i>|
+> |`loginCallBack`| Name of a callback method called after the authentication|
+> |`redirectUri`  |URL where users are sent after authentication against Azure AD v2 Endpoint|
+
+### Sign in users
+
+The following code snippet shows how you sign in users:
+
+```javascript
+userAgentApplication.loginRedirect(graphAPIScopes);
+```
+
+> |Where  |  |
+> |---------|---------|
+> |`graphAPIScopes`   | (Optional) Contains scopes being requested at login time (that is, `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs). |
+
+> [!TIP]
+> Alternatively, you may want to use the `loginPopup` method to display a popup window to sign in the user.
+
+### Requesting tokens
+
+MSAL has three methods used to acquire tokens: `acquireTokenRedirect`, `acquireTokenPopup` and `acquireTokenSilent`:
+
+#### Getting a user token silently
+
+The `acquireTokenSilent` method handles token acquisitions and renewal without any user interaction. After the `loginRedirect` or `loginPopup` method is executed for the first time, `acquireTokenSilent` is the method commonly used to obtain tokens that are used to access protected resources for subsequent calls - as calls to request or renew tokens are made silently.
+
+```javascript
+// Try to acquire the token used to query Graph API silently first:
+userAgentApplication.acquireTokenSilent(graphAPIScopes)
+```
+
+> |Where  |  |
+> |---------|---------|
+> |`graphAPIScopes`  | (Optional) Contains scopes being requested at login time (that is, `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs). |
+
+#### Getting a user token interactively
+
+ There are situations where you need to force users to interact with Azure AD v2.0 endpoint. For example:
+- Users may need to reenter their credentials because their password has expired
+- Your application is requesting access to a resource that the user needs to consent to
```

Comment:
Can we modify to "Your application is requesting access to additional resource scopes" since v2 is all about scopes mostly.